### DISCUSSION TITLE
如何提高emqx的订阅速率

### DISCUSSION THREAD

**Discussion by jiajiale** [emqx/emqx#4654] [Q&A] [answered]
使用阿里云Centos 2核4G部署emqx broker，同时用阿里云的另外一台服务器做客户端压测服务器，走内网通信，使用go创建协程的方式订阅emqx，创建10000个订阅，耗时9秒左右，平均1秒钟能成功订阅1千多，提高broker服务器配置到4核8G，同样创建10000个订阅，订阅速率没有明显增加，仍然只有1千多，broker的连接速率参数listener.tcp.external.max_conn_rate已经调整到5000，但是没有任何效果，请问如何提高订阅速率

**Comment by hjianbo** [comment, marked as answer]
Currently(below 4.3), the rate for topic subscription is a bottleneck in the system, especially establish a wildcard subscription in an emqx cluseter. We are trying to optimize it.

You can see the discussion here: https://github.com/emqx/emqx/pull/4646

**Comment by zmstone** [comment]
@jiajiale 
我们针对一下场景有性能优化正在进行：
* emqx集群
* wildcard 的主题
* 主题的非wildcard 前缀部分没有很多冲突（例如  `a/b/<uuid>/+/....`，这种在 + 前面部分，各个客户端之间不会冲突那种)
